scripts/zones/Spire_of_Dem/mobs/Progenerator.lua

```diff
@@ -32,18 +36,10 @@ entity.onMobFight = function(mob, target)
         mob:useMobAbility()
     end
 
-    if mob:getHPP() <= 35 then
-        mob:setMod(xi.mod.STORETP, 250)
-    end
-end
-
-entity.onMobDeath = function(mob, player, optParams)
-    local momma = mob:getID()
-    for i = momma + 1, momma + mob:getLocalVar('maxBabies') do
-        local baby = GetMobByID(i)
-        if baby and baby:isSpawned() then
-            baby:setHP(0)
-        end
+    if mob:getHPP() > 35 then
+        mob:setMod(xi.mod.REGAIN, 0)
+    elseif mob:getHPP() <= 35 then
+        mob:setMod(xi.mod.REGAIN, 100)
```

Comment:
Theres no need to define both conditions. They are mutually exclusive.

Also, whenever we fetch something more than once, its better to store it in a variable.